src/Vendor/Promotion/BasketPromotion.php

```diff
@@ -147,6 +163,42 @@ private static function denormalizePeriod(array $periodData): PromotionPeriod
         );
     }
 
+    private static function denormalizeTarget(array $targetData): BasketPromotionTarget
+    {
+        // We have to explode 'type' property because product_ids are serialized in it :(
+        $target = explode(';', $targetData['type']);
+        $type = $target[0];
+
+        $type = new BasketPromotionTargetType($type);
+
+        switch (true) {
+            case BasketPromotionTargetType::BASKET()->equals($type):
+                return new BasketTarget();
+
+            case BasketPromotionTargetType::PRODUCTS()->equals($type):
+                // We have to format products_ids data for ProductTarget constructor
+                if (is_string($target[1]) && $target[1] !== "") {
+                    $targetData['products_ids'] = explode(',', $target[1]);
```

Comment:
```suggestion
                    $targetData['products_ids'] = array_map(
                        function (string $id):int {
                            return (int) $id;
                        },
                        explode(',', $target[1])
                    );
```

Economie d'une variable :)

Reply:
:money_with_wings: